webactions.md

```diff
@@ -0,0 +1,546 @@
+< HTTP/1.1 200 OK
+< Server: nginx/1.11.13
+< Content-Length: 0
+< Connection: keep-alive
+< Access-Control-Allow-Methods: OPTIONS, GET
+< Access-Control-Allow-Origin: example.com
+```
+
+## Web Actions in Shared Packages
+
+A web action in a shared (i.e., public) package is accessible as a web action either directly via the package's fully qualified name, or via a package binding.
+
+**Important:** A web action in a public package is accessible for all bindings of the package even if the binding is private. This is because the web action annotation is carried on the action and cannot be overridden. If you don't want to expose a web action through your package bindings, clone-and-own the package instead.
+**[[NH: Note link to annotations.md in next paragraph.]]**
+Action parameters are inherited from its package and, if there is one, the binding. You can make package parameters [immutable](./annotations.md#protected-parameters) by defining their values through a package binding.
+
+## Error handling
+
+When a `nim` action fails, there are two different failure modes. The first is known as an _application error_ and is analogous to a caught exception: the action returns a JSON object containing a top level `error` property. The second is a _developer error_, which occurs when the action fails catastrophically and does not produce a response. This developer error is similar to an uncaught exception. For web actions, the controller handles application errors as follows:
+
+1. The controller projects an `error` property from the response object.
+1. The controller applies the content handling implied by the action extension to the value of the `error` property.
+
+Developers should be aware of how web actions might be used and generate error responses accordingly. A web action that is used with the `.http` extension should return an HTTP response, for example: `{error: { statusCode: 400 }`. Failing to specify the correct response will in a mismatch between the implied content-type from the extension and the action content-type in the error response. Special consideration must be given to web actions that are sequences, so that components that make up a sequence can generate adequate errors when necessary.
+
+## Vanity URL
+
+Web actions may be accessed via an alternate URL, which treats the Nimbella namespace as a subdomain to the API host. This is suitable for developing web actions that use cookies or local storage so that data is not inadvertently made visible to other web actions in other namespaces. The namespaces must match the regular expression `[a-zA-Z0-9-]+` (and should be 63 characters or fewer) for the edge router to rewrite the subdomain to the corresponding URI. For a conforming namespace, the URL `https://guest.openwhisk-host/public/index.html` becomes a alias for `https://openwhisk-host/api/v1/web/guest/public/index.html`. **[[NH: PLease advise on what to do with these URLs.]]**
```

Comment:
We have similar functionality in `nim` but not the same in detail.   To avoid confusion, the section needs to be more or less completely replaced with a section about how Nimbella generates URLs for customer namespaces.   I think either I or @rabbah needs to write the section.   You should probably just replace the section with a note to that effect.

Reply:
agree Josh or I will take it.

Reply:
Note added.